File: mobile/app/src/main/java/com/smartsuschef/mobile/network/dto/IngredientDtos.kt
package com.smartsuschef.mobile.network.dto

import com.google.gson.annotations.SerializedName

/**
 * Ingredient DTO
 * Maps to: IngredientDto(string Id, string Name, string Unit, decimal CarbonFootprint)
 */
data class IngredientDto(
    @SerializedName("id")
    val id: String,

    @SerializedName("name")
    val name: String,

    @SerializedName("unit")
    val unit: String,

    @SerializedName("carbonFootprint")
    val carbonFootprint: Double, // decimal in C#

    @SerializedName("createdAt")
    val createdAt: String,

    @SerializedName("updatedAt")
    val updatedAt: String
)

/**
 * Create Ingredient Request
 * Maps to: CreateIngredientRequest(string Name, string Unit, decimal CarbonFootprint)
 */
data class CreateIngredientRequest(
    @SerializedName("name")
    val name: String,

    @SerializedName("unit")
    val unit: String,

    @SerializedName("carbonFootprint")
    val carbonFootprint: Double // decimal in C#
)

/**
 * Update Ingredient Request
 * Maps to: UpdateIngredientRequest(string Name, string Unit, decimal CarbonFootprint)
 */
data class UpdateIngredientRequest(
    @SerializedName("name")
    val name: String,

    @SerializedName("unit")
    val unit: String,

    @SerializedName("carbonFootprint")
    val carbonFootprint: Double // decimal in C#
)

File: backend/SmartSusChef.Api/DTOs/IngredientDtos.cs
using System.ComponentModel.DataAnnotations;

namespace SmartSusChef.Api.DTOs;

public record IngredientDto(
    string Id,
    string Name,
    string Unit,
    decimal CarbonFootprint,
    string? GlobalIngredientId,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public record CreateIngredientRequest(
    [Required][StringLength(100)] string Name,
    [Required][StringLength(20)] string Unit,
    [Range(0, double.MaxValue)] decimal CarbonFootprint,
    string? GlobalIngredientId = null
);

public record UpdateIngredientRequest(
    [Required][StringLength(100)] string Name,
    [Required][StringLength(20)] string Unit,
    [Range(0, double.MaxValue)] decimal CarbonFootprint,
    string? GlobalIngredientId = null
);

// DTO for Global (immutable) ingredients
public record GlobalIngredientDto(
    string Id,
    string Name,
    string Unit,
    decimal CarbonFootprint,
    DateTime CreatedAt,
    DateTime UpdatedAt
);
